### PR TITLE
adding pairwise operator

### PR DESCRIPTION
**Pull request by tsouza**
This adds [pairwise](https://www.learnrxjs.io/learn-rxjs/operators/combination/pairwise) operator

### CODE REVIEW

File: lib/src/operators/pairwise.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  @override
+  void onNext(T value) {
+    if (previous != null) {
+      doNext(Pair(previous as T, value));
+    }
+    previous = value;
+  }
+}
+
+class Pair<T> {
```
Comment (renggli) [inline lib/src/operators/pairwise.dart:41]: Probably not needed in Dart 3 anymore (https://dart.dev/language/records), you can use (T, T)?

File: test/operators_test.dart [hunk 2]
```diff
@@ -5,6 +5,7 @@ import 'package:rx/core.dart';
 import 'package:rx/events.dart';
 import 'package:rx/operators.dart';
 import 'package:rx/schedulers.dart';
+import 'package:rx/src/operators/pairwise.dart';
```
Comment (renggli) [inline test/operators_test.dart:8]: This import shouldn't be necessary.

File: lib/src/operators/pairwise.dart [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+class PairwiseSubscriber<T> extends Subscriber<T> {
+  PairwiseSubscriber(Observer<Pair<T>> super.observer);
+
+  T? previous;
+
+  @override
+  void onNext(T value) {
+    if (previous != null) {
```
Comment (renggli) [inline lib/src/operators/pairwise.dart:34]: This does not work for nullable observables. You need to remember if you already have a value or not (look for example at how `AsyncSubject` does it).

**Review by renggli — COMMENTED** [review]
Great testing, thank you!

File: lib/src/operators/pairwise.dart (hunk 1 above)
Comment (tsouza) [reply lib/src/operators/pairwise.dart]: Sorry, I am relatively new to Dart and I forgot about this feature. So I defined it as:
```
typedef Pair<T> = (T first, T second);
```
Sounds good?

File: test/operators_test.dart (hunk 2 above)
Comment (tsouza) [reply test/operators_test.dart]: This was due to `Pair` not being exported. I've exported it now.

**Comment by renggli** [comment]
There are some checks (formatting) failing now, but otherwise looks great.

**Pull request merged**